contracts/SpokePool.sol

```diff
@@ -723,6 +871,60 @@ abstract contract SpokePool is
         );
     }
 
+    /**
+     * @notice An overloaded version of `speedUpV3Deposit` using `address` types for backward compatibility.
+     * This function allows the depositor to signal to the relayer to use updated output amount, recipient, and/or message
+     * when filling a deposit. This can be useful when the deposit needs to be modified after the original transaction has
+     * been mined.
+     *
+     * @dev The `depositor` and `depositId` must match the parameters in a `V3FundsDeposited` event that the depositor wants to speed up.
+     * The relayer is not obligated but has the option to use this updated information when filling the deposit using
+     * `fillV3RelayWithUpdatedDeposit()`. This version uses `address` types for compatibility with systems relying on
+     * `address`-based implementations.
+     *
+     * @param depositor The depositor that must sign the `depositorSignature` and was the original depositor.
+     * @param depositId The deposit ID to speed up.
+     * @param updatedOutputAmount The new output amount to use for this deposit. It should be lower than the previous value,
+     * otherwise the relayer has no incentive to use this updated value.
+     * @param updatedRecipient The new recipient for this deposit. Can be modified if the original recipient is a contract that
+     * expects to receive a message from the relay and needs to be changed.
+     * @param updatedMessage The new message for this deposit. Can be modified if the recipient is a contract that expects
+     * to receive a message from the relay and needs to be updated.
+     * @param depositorSignature The signed EIP712 hashstruct containing the deposit ID. Should be signed by the depositor account.
+     * If the depositor is a contract, it should implement EIP1271 to sign as a contract. See `_verifyUpdateV3DepositMessage()`
+     * for more details on how the signature should be constructed.
+     */
+    function speedUpV3Deposit(
+        address depositor,
+        uint32 depositId,
+        uint256 updatedOutputAmount,
+        address updatedRecipient,
+        bytes calldata updatedMessage,
+        bytes calldata depositorSignature
+    ) public {
+        _verifyUpdateV3DepositMessage(
+            depositor.toBytes32(),
+            depositId,
+            chainId(),
+            updatedOutputAmount,
+            updatedRecipient.toBytes32(),
+            updatedMessage,
+            depositorSignature,
+            UPDATE_V3_DEPOSIT_ADDRESS_OVERLOAD_DETAILS_HASH
```

Comment:
Why does the hash need to be different?

Reply:
Because, when adding backwards compatibility to `speedUpV3Deposit` with the address version, we still need to support an `updatedMessage` constructed with the old data type. Maybe this is unnecessary?

I was assuming that there might be cases were contracts are calling this `speedUpV3Deposit` with the old signature and thus passing updatedMessage built with the [old data structure](https://github.com/across-protocol/contracts/blob/231b5097bed59f17068b4f6fde916f940b423571/contracts/SpokePool.sol#L138-L141)

However if `speedUpV3Deposit` is only expected to be called off-chain we probably don't need backwards compatibility and can drop `UPDATE_V3_DEPOSIT_ADDRESS_OVERLOAD_DETAILS_HASH`.

Can you pls clarify how is `speedUpV3Deposit` is used @nicholaspai ?

Reply:
> However if speedUpV3Deposit is only expected to be called off-chain 

What do you mean by off-chain?

> Can you pls clarify how is speedUpV3Deposit is used @nicholaspai ?

The way this is used is a depositor sees a speed up button on the FE which prompts them to sign a message which they then submit to speedUpV3Deposit, code [here](https://github.com/across-protocol/frontend/blob/74a687cff46be5aad6f4a3d4fc65bef6dee130b8/src/views/Transactions/hooks/useSpeedUp.tsx#L86)